src/achievements/Achievement.ts

```diff
@@ -227,12 +243,14 @@ export abstract class TagUpAchievement extends Achievement<ts3.TagUp> {
 export abstract class TagDownAchievement extends Achievement<ts3.TagDown> {
     protected constructor(client: BotgartClient, imageURL: string, roleName: string, roleColour: string, repeatable: boolean, announceRepetitions: boolean) {
         super(client, imageURL, roleName, roleColour, repeatable, announceRepetitions);
+    }
 
-        client.ts3listener.on("tagdown", (x: ts3.TagDownEvent) => {
+    protected registerListeners() {
+        this.client.ts3listener.on("tagdown", async (x: ts3.TagDownEvent) => {
             if (x.commander.getDiscordMember() !== undefined) {
-                this.tryAward(<discord.GuildMember>x.commander.getDiscordMember(), x);
+                await this.tryAward(<discord.GuildMember>x.commander.getDiscordMember(), x);
```

Comment:
Same question here: wouldn't just leaving this as an asynchronous call be sufficient, as the result does not seem to be required at the call-site?

Reply:
This is deliberately done, so achievements are evaluated in a strict sequential order. 
Otherwise this line would just trigger a call to tryAward and continue with the next achievements.
Sequential processing should help finding achievements blocking the application resulting in it being terminated.

Reply:
I see! Unless `tryAward` has some sort of mutual exclusion mechanism (which I don't think it does) I am afraid this won't work the way you intend it to. The callback itself (line 249) is an `async` function, so each achievement will have its callback fired asynchronously, but within this asynchronous call, the result of `tryAward` will be awaited. Still, all achievements do this "in parallel".
If I remember correctly, `tryAward` should already be set up like:

```javascript
tryAward() {
  LOG.info("trying to award X");
  ...
  LOG.info("successfully awarded/ failed to award X");
}
```
which should give you an idea which achievements did not run to completion. If it is not set up this way, it probably should be.

Reply:
Actually, now that I think about it again, this probably _does_ work, due to the run-to-completion semantics of the job-queue.